File: Composite/cpp-source/ration/Breakfast.cpp
//
// Created by Pavel Akhtyamov on 2019-03-27.
//

#include <Breakfast.h>

size_t Breakfast::GetCalories() {
  return Dish::GetCalories() + 10;
}

std::shared_ptr<Dish> Breakfast::Add(std::shared_ptr<DishComponent> dish_component) {
  return Dish::Add(dish_component);
}

File: Composite/python-source/ration/Breakfast.py
from __future__ import annotations

from .Dish import Dish


class Breakfast(Dish):
    """Узел со своей добавкой к калориям.

    Показывает, что узел дерева может иметь собственное поведение сверх
    складывания частей: завтрак добавляет десять калорий на сервировку.
    """

    SERVING_CALORIES = 10

    def get_calories(self) -> int:
        return super().get_calories() + self.SERVING_CALORIES
